### PR TITLE
Applying CMS patches.

### PR DESCRIPTION
**Pull request by rappoccio**
Updating the 3.3.0 version of fastjet with our patches for thread safety and suppression of verbose printout.

### CODE REVIEW

**Comment by slava77** [comment]
1K lines seems more sensible than #14

**Comment by rappoccio** [comment]
Something seems a bit strange with the initial commit branch, it's giving aclocal errors:

```
cmslpc25:$ make
CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/sh /uscms/home/rappocc/nobackup/fastjet/cms-94x/fastjet/missing aclocal-1.15 -I m4
/uscms/home/rappocc/nobackup/fastjet/cms-94x/fastjet/missing: line 81: aclocal-1.15: command not found
WARNING: 'aclocal-1.15' is missing on your system.
         You should only need it if you modified 'acinclude.m4' or
         'configure.ac' or m4 files included by 'configure.ac'.
         The 'aclocal' program is part of the GNU Automake package:
         <http://www.gnu.org/software/automake>
         It also requires GNU Autoconf, GNU m4 and Perl in order to run:
         <http://www.gnu.org/software/autoconf>
         <http://www.gnu.org/software/m4/>
         <http://www.perl.org/>
Makefile:636: recipe for target 'aclocal.m4' failed
make: *** [aclocal.m4] Error 127
```

Was the 3.3.0 branch created from my branch or directly from the tarball?

**Comment by smuzaffar** [comment]
@rappoccio , cms/v3.3.0 branch was created from the tar bar  http://www.fastjet.fr/repo/fastjet-3.3.0.tar.gz

**Comment by rappoccio** [comment]
OK maybe aclocal needs to be updated at FNAL.

**Comment by smuzaffar** [comment]
@rappoccio , test faild to compile fastjet , see the PR tests build log here https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3732/26000/external/fastjet/3.3.0/log

File: include/fastjet/Error.hh [hunk 1]
```diff
@@ -89,12 +92,34 @@ private:
...
+/// class corresponding to critical internal errors
+/// 
+/// This is an error class (derived from Error) meant for serious,
+/// critical, internal errors that we still want to be catchable by an
+/// end-user [e.g. a serious issue in clustering where the end-user
+/// can catch it and retry with a different strategy]
+///
+/// Please directly contact the FastJet authors if you see such an
+/// error.
+class InternalError : public Error{
```
Comment (smuzaffar) [inline include/fastjet/Error.hh:118]: @rappoccio , looks like this class is defined multiple times.

**Comment by rappoccio** [comment]
Ok I will have to take a look Sunday or Monday, I am traveling today.

**Pull request merged**